syserol/model.py

```diff
@@ -126,3 +126,52 @@ def cross_validation():
         matrix[test_index, 6:13] = pred_matrix[test_index, len(glycan):len(glycan)+6] # store predicted values
     
     return matrix
+
+def class_predictions():
+    """ Predict Subject Class with Support Vector Machines and Decomposed Tensor Data"""
+    # Load Data
+    cube, glyCube = createCube()
+    classes = load_file("meta-subjects")
+    classes = classes.replace(to_replace=["controller", "progressor", "viremic", "nonviremic"], value = [1, 0, 1, 0])
+    cp = np.array(classes["class.cp"])
+    nv = np.array(classes["class.nv"])
+    
+    # Run Decomposition (5 components)
+    tensorFac, matrixFac, R2XX = perform_CMTF(cube, glyCube, 5)
+    # Subject Matrix acts for model training data
+    subjects_matrix = tensorFac[1][0]
+    
+    # Controller/Progressor classification
+    # Split data into training and testing data 
+    X_train, X_test, y_train, y_test = train_test_split(subjects_matrix, cp)
```

Comment:
You only have one CV fold? Should use `cross_val_predict` here.

Reply:
Okay let me know if the fixes below are how you wanted it!